CIP-0094/README.md

```diff
@@ -0,0 +1,350 @@
+
+For example:
+
+<table>
+<thead>
+  <th>CBOR diagnostic</td>
+  <th>Base16-encoded</th>
+</thead>
+<tbody>
+ <tr>
+  <td>
+<pre>
+{ 94:
+  { 0: [ "Pineapples on pizza?" ]
+  , 1:
+    [ [ "yes" ]
+    , [ "no" ]
+    ]
+  }
+}
+</pre>
+  </td>
+  <td>
+   <a target="_blank" href="https://cbor.me/?bytes=A1185EA200817450696E656170706C6573206F6E2070697A7A613F0182816379657381626E6F">
+<pre>
+A1185EA200817450696E656170706C6573206F6E
+2070697A7A613F0182816379657381626E6F
+</pre>
+   </a>
+  </td>
+ </tr>
+</tbody>
+</table>
+
+### Answer structure
+
+Similarly, an answer to a question is posted as transaction's metadata using the label `94` and the following metadata structure:
+
+```cbor
+answer =
+  { 2: question_hash
+  , 3: choice
+  , ? 4: vrf_witness
+  , ? 5: ed25519_witness
+  }
+
+question_hash =
+  bytes .size 32
+
+vrf_witness =
+  [ $vrf_vkey, vrf_proof ]
+
+vrf_proof =
+  [ bytes .size 64, bytes .size 16 ]
+
+ed25519_witness =
+  [ $vkey, $signature ]
+```
+
+Some remarks:
+
+1. The field `2` (`question_hash`) is a blake2b-256 hash digest, whose preimage is the entire serialised question metadata payload (with the `94` top-level label).
+1. The field `3` represents the 0-based index of the chosen answer from the available choices (from field `1` of the target poll).
+1. The fields `4` and `5` are not mutually optional; said differently, at least one of the two must be provided.
+    1. The `$vrf_vkey`, `$vkey` and `$signature` are as defined in the original [Cardano ledger specifications](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/babbage/test-suite/cddl-files/real/crypto.cddl) with the usual rules applying to transaction metadata (i.e. byte string longer than 64 bytes are encoded as lists)
+    1. In either case (VRF or Ed25519), the witness is calculated from the blake2b-256 hash digest of the serialised `answer` restricted to fields `2` and `3`.
+
+> **Warning**
+>
+> Note that the VRF here is **NOT** used as a digital signature algorithm, but used to produce a proof of possession of the corresponding VRF secret key. This is a _slight abuse_ of the VRF scheme since the verifiable random output produced is completely discarded, though safe provided that the VRF secret key was generated independently of the cold key.
+>
+> Let's stress this out a little: it is extremely important for implementors to keep that difference in mind and to **NOT** attempt to use the cold key as a VRF secret in an attempt to generate a VRF proof! **Ed25519 is only proven secure in the setting where an adversary has access to only signatures**. Attempting to re-use the same secret for different purpose could have dramatic consequences.
+>
+> See [Insecurity On Secret Key Re-Usage](https://github.com/input-output-hk/blogpost_breaking_ed25519_vrf/blob/main/blog.md#insecurity-of-secret-key-re-usage) for further detail.
```

Comment:
Maybe we want to link the Essential Cardano article, which contains the final version (I'm happy keeping any of the two)
https://www.essentialcardano.io/article/insecurity-of-secret-key-re-usage

Reply:
thanks @iquerejeta for noting the released version; I was wondering where that had ended up.

I agree the _Essential Cardano_ link would be a nice touch, but the GitHub blog page has better code syntax highlighting & has properly typeset maths instead of images.  Then again I've heard the GitHub maths don't work on mobile browsers, which would support using the Essential Cardano page.  Perhaps therefore they both can be linked?

Reply:
Sweet, thanks for pointing that out. The title suggested this had been posted somewhere but I also couldn't find where.